### PR TITLE
wine: include mpg123

### LINKED ISSUE #11592: Wine 2.4 (devel) fails to build
- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [x] Confirmed this is a problem with specific formulae and not Homebrew/brew? If it's a general Homebrew/brew problem please file this issue at https://github.com/Homebrew/brew/issues/new

When building wine with --devel, the install fails after printing a bunch of errors about unknown mpg123 symbols for i386. I think the problem is just the 32bit wine binary is trying to be built against a 64bit mpg123, and for some reason the --without-mpg123 build flag was removed from the configuration of 32bit wine in the formula. After manually adding --without-mpg123 to the configure args in the formula, the formula built successfully, but I don't know if this is the correct approach to be taking.

### PR DESCRIPTION
**Pull request by fxcoudert**
Trying to fix #11592

### CODE REVIEW

**Comment by fxcoudert** [comment]
Tests are good. @spaghetti2514 could you test this?

**Comment by spaghetti2514** [comment]
@fxcoudert Formula fails with
```Error: Permission denied - /usr/local/Cellar/wine/2.4_1/libexec/lib/libcrypto.1.0.0.dylib```
same as the test results

**Comment by spaghetti2514** [comment]
For the record, after applying the changes, reverting the new MachO::Tools.merge_machos sections, and just leaving the new mpg123 resource and build sections, wine builds correctly.

**Comment by fxcoudert** [comment]
The Ruby Mach-O code gives me this error and I do not really know why…
`Error: Permission denied - /usr/local/Cellar/wine/2.0_1/libexec/lib/libcrypto.1.0.0.dylib`

**Comment by JCount** [comment]
Not certain if that will help or not.

File: Formula/wine.rb [hunk 1]
```diff
@@ -239,22 +246,28 @@ def install
         system "make", "install"
 
         %w[libcrypto libssl].each do |libname|
-          system "lipo", "-create", "#{dirs.first}/#{libname}.1.0.0.dylib",
-                                    "#{dirs.last}/#{libname}.1.0.0.dylib",
-                         "-output", "#{libexec}/lib/#{libname}.1.0.0.dylib"
+          MachO::Tools.merge_machos("#{libexec}/lib/#{libname}.1.0.0.dylib",
+                                    %W[
+                                      #{dirs.first}/#{libname}.1.0.0.dylib
+                                        dirs.last}/#{libname}.1.0.0.dylib
```
Comment (woodruffw) [inline Formula/wine.rb]: This doesn't look quite right. Is `dirs.last}` here supposed to be `#{dirs.last}`?

**Comment by woodruffw** [comment]
I can't think of any reason off the top of my head why `merge_machos` would fail where `lipo` would succeed - @spaghetti2514, could you provide the full `--verbose` error output?

File: Formula/wine.rb (hunk 1 above)
Comment (JCount) [inline Formula/wine.rb]: oops

Comment (ilovezfs) [inline Formula/wine.rb]: why are we using an array anyway?

Comment (JCount) [inline Formula/wine.rb]: because I am apparently addled

**Comment by woodruffw** [comment]
@BrewTestBot test this please

**Comment by JCount** [comment]
Why is this so intermittent??

**Comment by woodruffw** [comment]
The Jenkins failures? Not sure.

**Comment by spaghetti2514** [comment]
I'm on a rather old laptop and it takes me quite a while to rebuild using the formula, so maybe someone else who can reproduce the issue can provide verbose output?

It may help to know that ```libcrypto.1.0.0.dylib``` generated by lipo has 755 permissions, while the one from merge_machos is 555

**Comment by fxcoudert** [comment]
@woodruffw testing locally, I get:

```
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/wine.rb:249:in `block (3 levels) in install'
NoMethodError: undefined method `merge_machos' for MachO::Tools:Module
```

A grep of `merge_machos` in `/usr/local/Homebrew/Library/Homebrew` show it is not present anywhere.

```
$ brew config
HOMEBREW_VERSION: 1.1.11-111-g00af5250
ORIGIN: https://github.com/Homebrew/brew
HEAD: 00af5250f0a7988178ed8d26520bf1a98a8dea9a
Last commit: 13 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 09ace045741f00850c8240c19494d739bdfcf9bf
Core tap last commit: 3 days ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: octa-core 64-bit haswell
Homebrew Ruby: 2.0.0-p648
Clang: 8.1 build 802
Git: 2.11.0 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Perl: /usr/bin/perl
Python: /Users/fx/anaconda3/bin/python => /Users/fx/anaconda3/bin/python3.5
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
Java: 1.6.0_65-b14-468
macOS: 10.12.4-x86_64
Xcode: 8.3
CLT: 8.3.0.0.1.1490382677
X11: 2.7.11 => /opt/X11
```

**Comment by fxcoudert** [comment]
I am reverting the Mach-O changes for now, want to get this PR in and fix wine. We can deal with calling or not calling cctools later.

**Comment by ilovezfs** [comment]
```
vendor/macho/macho/tools.rb
93:    def self.merge_machos(filename, *files)
```

**Comment by ilovezfs** [comment]
@fxcoudert 1.6.29 is out for libpng if you want to do that upgrade in this PR, too.

**Comment by fxcoudert** [comment]
Something's wrong here…

```
/usr/local/Homebrew/Library/Homebrew $ wc -l vendor/macho/macho/tools.rb
      84 vendor/macho/macho/tools.rb
```

**Comment by ilovezfs** [comment]
delete the whole vendor dir maybe and then try?

**Comment by fxcoudert** [comment]
The permission error is because openssl installs its dylibs without write permission:

```
/usr/local/Cellar/wine/2.0_1 $ ll libexec/lib/lib*.1.0.0.dylib
-r-xr-xr-x  1 fx  admin   1.4M Apr  2 13:52 libexec/lib/libcrypto.1.0.0.dylib
-r-xr-xr-x  1 fx  admin   338K Apr  2 13:52 libexec/lib/libssl.1.0.0.dylib
```

**Review by chenrui333 — APPROVED** [review]

**Pull request closed without merging**